### DISCUSSION TITLE
How are nightly builds published?

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/maui#21383] [Q&A] [answered]
I would like to test potential performance improvements thanks to https://github.com/dotnet/maui/commit/e0b85ca9fc65bd415efa0a0cff0f9f12c00578e9 but it looks like https://dev.azure.com/xamarin/public/_artifacts/feed/maui-nightly does not contain a nightly build containing the patch.

How are the nightly builds released? My expectation was that it's called nightly so that it's released every night (more or less) but that is certainly not true.

Thank you

**Comment by jfversluis** [comment, marked as answer]
Indeed every night, given that there have been changes merged that day. However, I think due to the rotation of some secrets etc. needed for publishing there was a disruption here. We're working on fixing this asap!

If you really want to test this now, or any other PR, for said PR go to the status checks, find the "MAUI-public" check and click Details and on that page click through to the Azure DevOps page. There you will see in the top bar "_n_ artifacts produces, _n_ consumed", click that and you will be able to download the produced artifacts that you can install locally.

For this PR the link is: https://dev.azure.com/xamarin/public/_build/results?buildId=111460&view=results

**Reply by MartyIX** [reply]
Thanks! 

The link you posted is very useful! It would be great to add it to https://github.com/dotnet/maui/wiki/Nightly-Builds so that people can test individual PR's against their codebases. That's something that is not probably so obvious for non-team members.
